core/src/main/java/hudson/ClassicPluginStrategy.java

```diff
@@ -591,7 +598,11 @@ private void updateTransientDependencies() {
         }
 
         private List<PluginWrapper> getTransitiveDependencies() {
-            if (transientDependencies==null) {
+          List<PluginWrapper> localTransientDependencies = transientDependencies;
```

Comment:
NIT: as mentioned above more likely this should have been `localTransitiveDependencies` as this is new code.